newIDE/app/src/ProjectsStorage/LocalProjectWriter.js

```diff
@@ -31,14 +38,61 @@ const writeJSONFile = (object: Object, filepath: string): Promise<void> => {
   }
 };
 
+const writeProjectFiles = (
+  project: gdProject,
+  newFilepath: string,
+  newProjectFolder: string
+): Promise<void> => {
+
+  const serializedProjectObject = serializeToJSObject(project);
+
+  if (project.isFolderProject()) {
+    const partialObjects = split(serializedProjectObject, {
+      pathSeparator: '/',
+      getArrayItemReferenceName: getSlugifiedUniqueNameFromProperty('name'),
+      shouldSplit: splitPaths(
+        new Set([
+          '/layouts/*',
+          '/externalLayouts/*',
+          '/externalEvents/*',
+          '/layouts/*',
+          '/eventsFunctionsExtensions/*',
+        ])
+      ),
+      isReferenceMagicPropertyName: '__REFERENCE_TO_SPLIT_OBJECT',
+    });
+
+    return Promise.all(
+      partialObjects.map(partialObject => {
+        return writeJSONFile(
+          partialObject.object,
+          path.join(newProjectFolder, partialObject.reference) + '.json'
+        ).catch(err => {
+          console.error('Unable to write a partial file:', err);
+          throw err;
+        });
+      })
+    ).then(() => {
+      return writeJSONFile(serializedProjectObject, newFilepath).catch(err => {
+        console.error('Unable to write the split project:', err);
+        throw err;
+      });
+    });
+  } else {
+    return writeJSONFile(serializedProjectObject, newFilepath).catch(err => {
+      console.error('Unable to write the project:', err);
+      throw err;
+    });
+  }
+};
+
 export default class LocalProjectWriter {
   static saveProject = (project: gdProject): Promise<void> => {
     const filepath = project.getProjectFile();
     const projectPath = path.dirname(project.getProjectFile());
     if (!filepath) {
       return Promise.reject('Unimplemented "Save as" feature');
     }
-
     const serializedProjectObject = serializeToJSObject(project);
```

Comment:
You can now remove this line and the rest until the end of the function, and replace it by `return writeProjectFiles(project, filepath, projectPath);` ;)
Make sure to test that save still work after doing this though ;)